sha3/src/main/java/com/theromus/sha/Keccak.java

```diff
@@ -62,23 +192,22 @@
 
         // Squeezing phase
         ByteArrayOutputStream byteResults = new ByteArrayOutputStream();
-        int tOutputLen = parameter.getOutputLen() / 8;
+        int tOutputLen = (int) Math.ceil(customLen / 8.0);
+        byteResults.write(0);
         while (tOutputLen > 0) {
             blockSize = min(tOutputLen, rateInBytes);
-            for (int i = 0; i < blockSize; i++) {
+            for (int i = 0; i < blockSize; i++)
                 byteResults.write((byte) uState[i]);
-            }
 
             tOutputLen -= blockSize;
-            if (tOutputLen > 0) {
+            if (tOutputLen > 0)
                 doKeccakf(uState);
-            }
         }
 
-        return byteResults.toByteArray();
+        return cutBit(byteResults.toByteArray(), (int) ((Math.ceil(customLen / 8.0)*8) - customLen));
     }
 
-    private void doKeccakf(final int[] uState) {
+    private static void doKeccakf(final int[] uState) {
```

Comment:
Convert to non-static.

Reply:
Why? isn't better leave this function static?

Reply:
Ok, let's leave this static-signature.